src/components/atoms/loadingWrapper/LoadingWrapper.tsx

```diff
@@ -1,42 +1,93 @@
-import React, { useState, HTMLAttributes } from 'react';
+import React, { useState, HTMLAttributes, useEffect } from 'react';
 import Loading from '../loading/Loading';
 import './loadingWrapper.scss';
 
 export interface LoadingWrapperInterface
   extends HTMLAttributes<HTMLDivElement> {
   data: any;
   animation?: boolean;
+  // option to make animation trigger after n milliseconds
+  startAfter?: number;
   animationTime?: number;
 }
 
 // TODO smoother rendereing of component
+// Render components after data is fetched
+// animationTime -> ensures that the loading animation is displayed for minimum "animationTime" milliseconds (only if animation is true)
+// startAfter -> if > 0 the render animation starts after "startAfter" milliseconds (only if animation is true)
 const LoadingWrapper: React.FC<LoadingWrapperInterface> = ({
   data,
-  animation = false,
+  animation = true,
   animationTime = 250,
+  startAfter = 0,
   children,
   ...rest
 }) => {
   // sets render to opposite to animation since animation true will need to explicit set
   // render for childeren to load and no animation will just render children when defined
+  const [isFetched, setIsFetched] = useState<boolean>(false);
+  const [animationRender, setAnimationRender] = useState<boolean>(false);
+  const [startAfterDone, setStartAfterDone] = useState<boolean>(
+    startAfter === 0
+  );
   const [render, setRender] = useState<boolean>(false);
+  const [animationDone, setAnimationDone] = useState<boolean>(
+    animation === false
+  );
 
-  var interval = setInterval(function () {
+  function startAfterTimeout() {
+    setTimeout(function () {
+      setAnimationRender(true);
+    }, startAfter);
+  }
+
+  function startAnimationTimout() {
+    setTimeout(function () {
+      setAnimationDone(true);
+    }, animationTime);
+  }
+
+  useEffect(() => {
     if (data != null) {
+      setIsFetched(true);
+      if (!startAfterDone) {
+        // allows rendering if data is fetched and animation has not started
+        setRender(true);
+      }
+    }
+  }, [data]);
+
+  useEffect(() => {
+    if (animationDone && isFetched) {
       setRender(true);
-      clearInterval(interval);
     }
-  }, animationTime);
+  }, [isFetched, animationDone]);
+
+  useEffect(() => {
+    if (!animation) {
+      return;
+    }
+    startAnimationTimout();
+  }, [animationRender]);
+
+  useEffect(() => {
```

Comment:
Since I can't see any usage of the setter, then we probably don't need the useEffect since the value never changes